### DISCUSSION TITLE
请问是否可以新构造一条合并转发消息?

### DISCUSSION THREAD

**Discussion by Alex1911-Jiang** [Hoshikawa-Kaguya/Sora#58] [Q&A] [answered]
不存在原消息, 里面的内容都是全新的, 但是以合并转发的方式发送
看了一下 ForwardSegment 似乎不能修改转发消息的内容
其他的机器人可以这样凭空构造, 不知道在这该怎么写, 求指导

**Comment by Yukari316** [comment, marked as answer]
可以试一试使用 `CustomNode` 完成消息的构建
每一个 `CustomNode` 就是一个消息节点
每一个消息节点可以是一条别人发送的消息，或者是自己凭空生成的消息，也可以是另一个合并转发套娃
具体的使用方法请参考gocq的文档：[合并转发消息节点](https://docs.go-cqhttp.org/cqcode/#%E5%90%88%E5%B9%B6%E8%BD%AC%E5%8F%91%E6%B6%88%E6%81%AF%E8%8A%82%E7%82%B9)

这个是我自己在机器人中构建消息并发送的一个示范：
```csharp
var customNodes = new List<CustomNode>();
for (int i = 0; i < count; i++)
{
    customNodes.Add(new CustomNode("色色", 114514, SoraSegment.Image($"{imageUrl}/{i}")));
}
await eventArgs.SourceGroup.SendGroupForwardMsg(customNodes);
```
这是效果：
![色色](https://user-images.githubusercontent.com/7535224/169664899-d8be182f-5e98-424b-a9c6-cc1b487660ec.jpg)
